corehq/apps/locations/permissions.py

```diff
@@ -164,12 +164,24 @@ class LocationResource(HqBaseResource):
     return resource
 
 
+def conditionally_location_safe(conditional_function):
+    """Decorator to apply to a view function that verifies if something is location
+    safe based on the arguments or kwarguments. That function should return
+    True or False.
+
+    """
+    def _inner(view_fn):
+        view_fn._conditionally_location_safe_function = conditional_function
+        return view_fn
+    return _inner
+
+
 def location_restricted_response(request):
     from corehq.apps.hqwebapp.views import no_permissions
     return no_permissions(request, message=LOCATION_ACCESS_DENIED)
 
 
-def is_location_safe(view_fn, view_args, view_kwargs):
+def is_location_safe(view_fn, view_args, view_kwargs, request=None):
```

Comment:
You are right that it currently isn't used, however all view functions currently receive the `request` method, and I was finding it confusing that the signature for the conditionally location safe function didn't include the request. I also thought it might be useful if ever you want to make sure if a `request.couch_user` has a specific flag or permission or something. I guess this might be an instance of YAGNI. I'll remove it for now, if anyone wants it in the future, they can update this method themselves.